### DISCUSSION TITLE
Acutal JSON schema or Open Api JSON schema

### DISCUSSION THREAD

**Discussion by lud-wj** [OAI/sig-moonwalk#65] [Q&A]
I am not sure this is written somewhere, it seems it is not.

Does this proposal relies on actual  JSON Schemas or the incompatible ones used in version 3 where for instance you cannot tell `"type": ["string", "object", "null"]` but have to provide `"nullable": true, "oneOf": ...`.

The wording tells "JSON Schema" but I know that the same wording is used everywhere when the schemas are actually not proper JSON schemas. So it could be worth  make that explicit.

Thank you.

**Comment by handrews** [comment]
OAS 3.1 already uses "actual" JSON Schema 2020-12 (with a few extensions that for validation can be safely ignored, but can impact code gen, e.g. `discriminator`).

I cannot imagine Moonwalk going backwards on this, but as the recent blog post about shipping Moonwalk in 2024 notes, thre is also a desire to loosen the coupling between OAS and any specific schema system (which I agree with as a goal).  My guess is that it wil retain the ability to use "actual" JSON Schema (whatever that means in 2024- it's [not entirely clear](https://github.com/orgs/json-schema-org/discussions/557) where JSON Schema is headed these days), but may support alternatives.  But that's just a guess on my part.

**Reply by lud** [reply]
Oh gosh I'm sorry I totally missed that 3.1 had fixed that. It seems that despite being maintained, the tool we use is stuck on 3.0.

Well that could be good to allow different schema forms but then you need a list of algorithms and formats shared by everyone...

Anyway, thank you for the pointers, very interesting!

**Reply by lud-wj** [reply]
(wrong account sorry :D)

**Comment by handrews** [comment]
Closing since this was answered and has no Moonwalk impact.
